src/Registration.php

```diff
@@ -5,7 +5,7 @@
 
 use OutOfBoundsException;
 
-class Registration
+class Registration implements RegistrationInterface
```

Comment:
There are a couple of typehints in `Server` that would also benefit from being updated to accept the interface. Would you mind updating those too?